contrib/mesos/pkg/controllermanager/controllermanager.go

```diff
@@ -172,12 +172,16 @@ func (s *CMServer) Run(_ []string) error {
 	}
 
 	if s.AllocateNodeCIDRs && s.ConfigureCloudRoutes {
-		routes, ok := cloud.Routes()
-		if !ok {
-			glog.Fatal("Cloud provider must support routes if configure-cloud-routes is set")
+		if cloud == nil {
+			glog.Warning("configure-cloud-routes is set, but no cloud provider specified. Will not configure cloud provider routes.")
+		} else if routes, ok := cloud.Routes(); !ok {
+			glog.Warning("configure-cloud-routes is set, but cloud provider does not support routes. Will not configure cloud provider routes.")
+		} else {
+			routeController := routecontroller.New(routes, clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "route-controller")), s.ClusterName, clusterCIDR)
+			routeController.Run(s.NodeSyncPeriod.Duration)
 		}
-		routeController := routecontroller.New(routes, clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "route-controller")), s.ClusterName, clusterCIDR)
-		routeController.Run(s.NodeSyncPeriod.Duration)
+	} else {
```

Comment:
why not just copy/paste the else conditions from the upstream code. looks like you just combined the error messages here.